From: David Bauer <mail@david-bauer.net>
Subject: [PATCH mt76] wifi: mt76: mt7915: set mt76 specific PS flag
Date: Fri, 13 Mar 2026 12:25:00 +0100

mt76 tracks the PSM state of a sta internally with a wcid flag. TX to
such clients is skipped based on the presence of this flag.

This flag was not added to the PS state notify handler for MT7915 chips.
Without this flag, mt76 queues pending frames to the hardware,
accounting for airtime when a PSM notification is received while in a TX
iteration.

Set the PS flag for the STA WCID to prevent this from happening. TX gets
skipped in presence of this flag.

Signed-off-by: David Bauer <mail@david-bauer.net>
---
 mt7915/mcu.c | 5 +++++
 1 file changed, 5 insertions(+)

--- a/mt7915/mcu.c
+++ b/mt7915/mcu.c
@@ -476,6 +476,11 @@ static void mt7915_mcu_rx_ps_sync(struct
 	if (!sta)
 		goto out;
 
+	if (p->ps_bit)
+		set_bit(MT_WCID_FLAG_PS, &wcid->flags);
+	else
+		clear_bit(MT_WCID_FLAG_PS, &wcid->flags);
+
 	ieee80211_sta_ps_transition_ni(sta, !!p->ps_bit);
 
 out:
